InvenTree/InvenTree/tasks.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import re
+import json
+import requests
+import logging
+
+from datetime import datetime, timedelta
+
+from django.core.exceptions import AppRegistryNotReady
+from django.db.utils import OperationalError, ProgrammingError
+
+
+logger = logging.getLogger(__name__)
+
+
+def schedule_task(taskname, **kwargs):
+    """
+    Create a scheduled task.
+    If the task has already been scheduled, ignore!
+    """
+
+    # If unspecified, repeat indefinitely
+    repeats = kwargs.pop('repeats', -1)
+
+    try:
+        from django_q.models import Schedule
+    except (AppRegistryNotReady):
+        logger.warning("Could not start background tasks - App registry not ready")
+        return
+
+    try:
+        if Schedule.objects.filter(func=taskname).exists():
```

Comment:
So what happens when the task parameters change?   Like you want to shift the period from 5 minutes to 30 minutes say?

Reply:
Good point. Code has been updated now.